plugins/chat/lib/chat/channel_hashtag_data_source.rb

```diff
@@ -23,7 +23,7 @@ def self.channel_to_hashtag_item(guardian, channel)
     end
 
     def self.lookup(guardian, slugs)
-      if SiteSetting.enable_experimental_hashtag_autocomplete
+      if SiteSetting.enable_experimental_hashtag_autocomplete && SiteSetting.enable_public_channels
```

Comment:
Kind of annoying we can't just do this in plugin.rb but I saw you discuss with David (and remember stuff in the past) for why we can't. After you merge this PR I think I will make a change to the API for hashtag data sources to add an `enabled?` method to them that can have stuff like this, so you don't need to add it on every line.

Reply:
Yes agreed, but it's okayish